src/index.js

```diff
@@ -25,7 +26,11 @@ Walker.prototype._read = function () {
   self.fs.lstat(pathItem, function (err, stats) {
     var item = { path: pathItem, stats: stats }
     if (err) return self.emit('error', err, item)
-    if (!stats.isDirectory()) return self.push(item)
+
+    if (!stats.isDirectory() || (self.rootDepth &&
+      pathItem.split(path.sep).length - (self.rootDepth + 1) >= self.options.depthLimit)) {
```

Comment:
Sorry for being late here! @RyanZim that's a good argument. However, don't you think using `path.normalize()` is somehow redundant, since we use `path.resolve()` on root directory and `path.join()` on its subdirectories, and both methods normalizing the resulting path anyway?

https://nodejs.org/api/path.html#path_path_join_paths
https://nodejs.org/api/path.html#path_path_resolve_paths